WarshipGame/Assets/Scripts/Gameplay/Grid/GlowManager.cs

```diff
@@ -0,0 +1,118 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Manages material colors and shaders to be activated and changed on selection.
+/// </summary>
+public class GlowManager : MonoBehaviour
+{
+    private readonly Dictionary<Renderer, Material[]> _selectionMaterialDictionary = new Dictionary<Renderer, Material[]>();
+    private readonly Dictionary<Renderer, Material[]> _originalMaterialDictionary = new Dictionary<Renderer, Material[]>();
+    private readonly Dictionary<Color, Material> _cachedSelectionMaterials = new Dictionary<Color, Material>();
+    private readonly Color _validHexColor = Color.green;
+    private Color _originalColor;
+    
+    public Material SelectionMaterial;
+    public bool IsGlowing = false;
+    
+    private void Awake()
+    {
+        PrepareMaterialDictionaries();
+        _originalColor = SelectionMaterial.GetColor("_GlowColor");
+    }
+
+    /// <summary>
+    /// Fills in material dictionaries for later use 
+    /// </summary>
+    private void PrepareMaterialDictionaries()
+    {
+        foreach (Renderer renderer in GetComponentsInChildren<Renderer>())
+        {
+            Material[] originalMaterials = renderer.materials;
+            _originalMaterialDictionary.Add(renderer, originalMaterials);
+            Material[] newMaterials = new Material[renderer.materials.Length];
+
+            for (int i = 0; i < originalMaterials.Length; i++)
+            {
+                Material mat = null;
+                if (!_cachedSelectionMaterials.TryGetValue(originalMaterials[i].color, out mat))
```

Comment:
`Material mat = null;` is redundant and can be added to the if statement by doining `out Material mat`

Reply:
"i do not depend on my code editor to fix my code at all" 
*insert clown emoji*